### DISCUSSION TITLE
Feature Request: Allow list[str] for stubPath setting

### DISCUSSION THREAD

**Discussion by Jasha10** [microsoft/pylance-release#2658] [Enhancement]
Currently the `stubPath` setting takes a string (indicating which directory contains stubs).
I'd like to pass a list of strings (indicating which directories contain stubs).

**Motivation:**
Some of my stubs are automatically generated by computers, and some of my stubs are coded by hand. I'd like to keep these sets of stubs in separate directories.

**Describe the solution you'd like**
It would be nice to be able to set `stubPath = ["generated", "stubs"]` in my `pyproject.toml` file.

**Additional context**
`$PYTHONPATH` and `$MYPYPATH`, which are comparable to `stubPath`, allow multiple directories.

**Comment by erictraut** [comment]
Thanks for the feature request. This is the first time we've received a request for this use case. This sounds like it might be pretty specific to your situation. This change would impact not only pyright but also pylance and its settings and documentation, so I'm going to transfer the feature request to the pylance project. We'll discuss it there and see if we get signal from other pylance and pyright users that this is something that would be more broadly desired.

I'll note that the pyright configuration setting file was loosely modeled after the typescript configuration settings, and typescript does support multiple "typing roots" (the equivalent of type stubs). So there is some precedence for this.

**Reply by jdtsmith** [reply]
It would also be nice to set `stubPath = ["/path/to/common/stub-directories", "./typings"]` or similar, so you don't have to link commonly used stubs into project directories.

**Comment by judej** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes.

**Comment by aholmes** [comment]
Hello,

I have a use case in a mono-repo. The root of the repository allows for working with individual packages under a `src/<package>/` directory. Each of these packages is their own independent package, and can be worked on without the mono-repo. So for example, one developer may want to work on both `src/web/` and `src/database/` with a venv in the root of the mono-repo and editable installs of these packages, and another developer may want to work on just `src/web/` with a venv in the `src/web/` directory.

Each package has their own independent dependencies defined in a `pyproject.toml` file under their respective `src/<package>/` directory. As such, it is ideal to store type stubs for those dependencies under `src/<package>/typings/` rather than the root of the mono-repo at `typings/`. The mono-repo itself uses its own type stubs as well, so also uses the `typings/` directory.

So in short, I can currently store type stubs for every independent package in a mono-repo's root `typings/` directory, which couples independent packages during development. It would be nice to specify:

```toml
stubPath = [
    "./typings",
    "./src/web",
    "./src/database"
]
```
